Namitha - fix(JobAnalytics): prevent invalid date range render + react to dark

Namitha7070 opened the pull request:
**Description**

Page: Job Posting Page Analytics
Goal: Create a grouped/stacked horizontal bar chart showing Hits vs Applications per role, and address reviewer feedback.

Fixes (PRIORITY HIGH):
Overlap on small screens: Chart and date controls now reflow without colliding.
Dark mode not updating: UI and chart adopt the active theme correctly.
Dates UX: Replaced single dropdown with two date inputs (Start, End).
Overflow: Graph no longer spills outside its container.

## Related PRS (if any):
…

## Main changes explained:
1. Dates UX (range-first)**
  * Replaced the single *Dates* dropdown with two inputs: **Start date** / **End date**.
  * Filtering applies **date range first**, then role filter.
  * Added a **Clear** action to reset both dates and results quickly.

2. Invalid-range handling**
  * If **Start > End**, show an **inline warning banner** and **skip filtering/fetching**.
  * Chart data is cleared in this state; banner auto-clears once dates become valid.

3. Chart restoration (Recharts, horizontal)**
  * Restored the original visual spec: **gridlines, axes, end-of-bar value labels**.
  * Sorted roles **least → most popular** for quick scanning.
  * Preserved tooltips/labels; long role names truncate with a title/tooltip.

4. Overflow & responsiveness**
  * Clamped bar widths and contained the chart so it **stays inside the card**.
  * Adopted a **stable flex layout** for chart vs. filters to prevent overlap on small screens.
  * Reserved axis/label space to avoid clipping; no horizontal scroll.

5. Dark mode**
  * Introduced **theme CSS variables** and supported common selectors: `html.dark`, `[data-theme="dark"]`, `body.dark`.
  * Updated **bar, grid, axis, and text** colors to switch with the theme.
  * Set chart title to a **light blue** in dark mode for contrast.

6. Title & typography**
  * Updated page title to **“Least popular roles”** and aligned typography/spacing with the spec.

7. Data prep & state**
  * Refactored with **`useMemo`** for roles list, grouped counts, and max calculations.
  * Removed dead dropdown logic and **simplified state** for reliability.

## How to test:
check into current branch: git checkout fix/analytics-date-warning-darkmode
do npm install and ... to run this PR locally
Clear site data/cache
log as admin user
go to (http://localhost:5173/applicant-analytics)
Verify the below functions:
Dark mode: Toggle light↔dark
Invalid range: Set Start > End → warning shows
Valid range: Fix dates → warning clears; data renders.
Role filter: Clear all roles → chart clears; no errors. 
Minimize the window size to check if the chart is collapsing or not on smaller screens.

## Screenshots or videos of changes:
https://github.com/user-attachments/assets/04eab037-9de1-4f70-8858-51d464b77b68

Namitha7070 committed:
fix(JobAnalytics): prevent invalid date range render + react to dark theme

sohamsharma08 reviewed (approved):
Tested the PR using an admin account. The graph, warnings, and data is portraying as expected and everything is working fine. Relevant screenshots are attached below:

<img width="1297" height="703" alt="PR 3994 (1)" src="https://github.com/user-attachments/assets/c1ea60cd-1044-4310-ad5e-fb99b4df6e23" />
<img width="1297" height="703" alt="PR 3994 (2)" src="https://github.com/user-attachments/assets/d149b55f-3dc8-4f3e-90f6-f63dd1173c91" />
<img width="1297" height="703" alt="Pr 3994 (3)" src="https://github.com/user-attachments/assets/a8b1c5ba-8f85-4e34-8177-c6ac439b7575" />
<img width="1297" height="703" alt="PR 3994 (4)" src="https://github.com/user-attachments/assets/44ba43f4-369c-4c12-b6bb-9776be99e07e" />

vivekchandrabs reviewed (dismissed):
<img width="1678" height="713" alt="Screenshot 2025-09-04 at 3 33 36 PM" src="https://github.com/user-attachments/assets/92d5d007-22f3-484c-81d2-19673e85a945" />
<img width="1665" height="741" alt="Screenshot 2025-09-04 at 3 33 49 PM" src="https://github.com/user-attachments/assets/f229f3f1-3191-4515-ba24-c52922d6ca6b" />
<img width="1672" height="724" alt="Screenshot 2025-09-04 at 3 34 22 PM" src="https://github.com/user-attachments/assets/b2aa5b3c-6430-4894-b9f1-4ab532994b90" />

I checked all the functionality and everything is working as expected. 

@Namitha7070 Please document the PR properly. The frontend URL is /application-analytics

ezreal2020 reviewed (approved):
Verified n tested, everything worked as expected.

<img width="1456" height="748" alt="image" src="https://github.com/user-attachments/assets/d2fcf2fa-855d-467e-9ced-7fae03a7fa2e" />
<img width="1050" height="636" alt="image" src="https://github.com/user-attachments/assets/bb533cd5-f5e8-477f-a914-dd2b9a4928dd" />

munishpatel reviewed (commented):
I am getting Page Not Found in the link, should I change any thing locally to review this pull request?
<img width="1272" height="977" alt="Screenshot 2025-09-14 at 12 36 09 AM" src="https://github.com/user-attachments/assets/35ec690c-dbfe-48c1-a6fe-0309bccf09dd" />

Anusha-Gali reviewed (changes requested):
Hi Namitha,

I have reviewed your PR locally and though it works as per requirement. Below are the UI issues:
- The calendar is in light mode when the application is in dark mode
- The clear button is not within the browser in desktop as well as mobile

<img width="1272" height="824" alt="Image" src="https://github.com/user-attachments/assets/c829fdbc-0c29-47b9-932c-a32a2eb0002f" />
<img width="1278" height="824" alt="Image" src="https://github.com/user-attachments/assets/227758c5-1a6c-4011-a3a3-6a927a1b6038" />
<img width="1278" height="824" alt="Image" src="https://github.com/user-attachments/assets/c1b0dbb4-87ca-4cd0-be66-966d5e7a6f88" />
<img width="1278" height="824" alt="Image" src="https://github.com/user-attachments/assets/ecd18880-f191-43b3-bd56-e4dc4d3de46e" />
<img width="1278" height="824" alt="Image" src="https://github.com/user-attachments/assets/14dadb9c-f490-432b-877e-1f9836cb4556" />
<img width="1278" height="824" alt="Image" src="https://github.com/user-attachments/assets/ab802a77-f047-47f6-a050-82362986cef8" />

HemanthNidamanuru reviewed (changes requested):
Hi Namitha,
I have reviewed your PR locally and the functionality works as per requirement, however I noticed a few issues:

1. The calendar is in light mode when the application is in dark mode
2. The clear button is not visible in the desktop version but is visible in mobile
3.  I am able to select future dates

<img width="1898" height="1036" alt="Image" src="https://github.com/user-attachments/assets/b9294cec-4114-41db-9d83-28ff04b016eb" />
<img width="1895" height="1035" alt="Image" src="https://github.com/user-attachments/assets/d205a532-a5fb-4ac3-944f-7c821982db87" />
<img width="1900" height="1086" alt="Image" src="https://github.com/user-attachments/assets/ee219aec-ee6e-4663-98ed-f731f4a65bdc" />
<img width="642" height="1030" alt="Image" src="https://github.com/user-attachments/assets/b96bc671-fb5c-4bf2-88e5-5bfe05688e97" />

Namitha7070 committed:
fix: dark mode calendar and Clear button layout in applicant analytics

Namitha7070 committed:
fix: dark mode calendar and Clear button layout in applicant analytics

Namitha7070 committed:
Merge remote-tracking branch 'origin/development' into fix/Jobanalytics-date-warning-darkmode

# Conflicts:
#	src/components/ApplicationAnalytics/index.jsx
#	src/components/ApplicationAnalytics/jobAnalytics.jsx
#	src/components/ApplicationAnalytics/jobAnalytics.module.css

rohanrastogi311 reviewed (dismissed):
Hi Namitha, 

Well done with this implementation. 

<img width="1727" height="1038" alt="Image" src="https://github.com/user-attachments/assets/44d0042c-fa37-4951-b6b5-1ded5b877c0c" />

Anusha-Gali reviewed (changes requested):
Hi Namitha,

Thank you for fixing the dark mode issues and the clear button however i think the mobile responsiveness got disturbed as you can see it does collapse bit goes out of the layout and the x axis points combine to form a numerical number series. Also if you could increase the brightness of the warning message it would be great.

<img width="1269" height="754" alt="Image" src="https://github.com/user-attachments/assets/36664c3d-88e3-4fcc-aab7-91fcb08705ac" />

**Issues**
<img width="1269" height="754" alt="Image" src="https://github.com/user-attachments/assets/c80a4d85-bbf8-4dfb-883a-7615db027ee4" />
<img width="1269" height="754" alt="Image" src="https://github.com/user-attachments/assets/316a8a85-8a32-4e0e-9345-f20d2e4ed1e4" />
<img width="1269" height="754" alt="Image" src="https://github.com/user-attachments/assets/834a8ecc-5c20-4896-bbe9-1e132cddcd26" />

Namitha7070 committed:
fix: improve mobile responsiveness and dark mode visibility for job analytics

Namitha7070 commented:
> Hi Namitha,
> 
> Thank you for fixing the dark mode issues and the clear button however i think the mobile responsiveness got disturbed as you can see it does collapse bit goes out of the layout and the x axis points combine to form a numerical number series. Also if you could increase the brightness of the warning message it would be great.
> 
> <img alt="Image" width="1269" height="754" src="https://private-user-images.githubusercontent.com/143547248/568160552-36664c3d-88e3-4fcc-aab7-91fcb08705ac.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NzUyNDM3ODMsIm5iZiI6MTc3NTI0MzQ4MywicGF0aCI6Ii8xNDM1NDcyNDgvNTY4MTYwNTUyLTM2NjY0YzNkLTg4ZTMtNGZjYy1hYWI3LTkxZmNiMDg3MDVhYy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjYwNDAzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI2MDQwM1QxOTExMjNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1iYWMyMjIyYzk0YzczOTIzZDcxMDAxODljMWJmOThjNzJjMDE2Zjk3MTM2ZDE4NmZkOGYwYWM2OTI2YzI5Y2M4JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.G8bfy3jr_otPj_RSNbO_3vMK6mFyD4B9JOI1rz-RSK4">
> **Issues** <img alt="Image" width="1269" height="754" src="https://private-user-images.githubusercontent.com/143547248/568160549-c80a4d85-bbf8-4dfb-883a-7615db027ee4.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NzUyNDM3ODMsIm5iZiI6MTc3NTI0MzQ4MywicGF0aCI6Ii8xNDM1NDcyNDgvNTY4MTYwNTQ5LWM4MGE0ZDg1LWJiZjgtNGRmYi04ODNhLTc2MTVkYjAyN2VlNC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjYwNDAzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI2MDQwM1QxOTExMjNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1iOWE5MjQzOTU4NjhiMDgwN2M2MTc2MDdkMjU2MTU5NGI4ZTFiN2ZjOTM4NmYzNTViNmZiMWI2ZjA1MGZlMjBkJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.Y6vX3vcxWzj55eE6F7psP2QX2LJKybXiIOxoXiX3DKQ"> <img alt="Image" width="1269" height="754" src="https://private-user-images.githubusercontent.com/143547248/568160550-316a8a85-8a32-4e0e-9345-f20d2e4ed1e4.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NzUyNDM3ODMsIm5iZiI6MTc3NTI0MzQ4MywicGF0aCI6Ii8xNDM1NDcyNDgvNTY4MTYwNTUwLTMxNmE4YTg1LThhMzItNGUwZS05MzQ1LWYyMGQyZTRlZDFlNC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjYwNDAzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI2MDQwM1QxOTExMjNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT01Mzc2YWZmNjQ1NWI3Yjc2NTgwMDE4NTljOWI4ZDVkNTY1NTk2MmNmZWZjMDdmYzY2ZDA3OGViY2QxNTkxYjBmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.MU71FZWg3cFs-Yzs5lVe5eaE48D6_JM3wiq4DJmkd3s"> <img alt="Image" width="1269" height="754" src="https://private-user-images.githubusercontent.com/143547248/568160548-834a8ecc-5c20-4896-bbe9-1e132cddcd26.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NzUyNDM3ODMsIm5iZiI6MTc3NTI0MzQ4MywicGF0aCI6Ii8xNDM1NDcyNDgvNTY4MTYwNTQ4LTgzNGE4ZWNjLTVjMjAtNDg5Ni1iYmU5LTFlMTMyY2RkY2QyNi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjYwNDAzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI2MDQwM1QxOTExMjNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT02NDdkMjliZmJkODJiZTBlY2NiMTIyYzAzMDQ5M2RkNGEzMDgzOTVmODg2NDFlMTcyN2M1OWQwMmUxODc2ZTUzJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.pNC8uKukKHNlFwshTl1KoEvBO7arilJFU9wU5_z24lY">

Namitha7070 commented:
Hi Anusha, I have fixed the mobile responsiveness of the page and fixed the x-axis and also the "No data for the selected filters." message is brighter, more visible.
<img width="1918" height="944" alt="Screenshot 2026-04-03 113050" src="https://github.com/user-attachments/assets/0c3dd638-38df-4c02-ba7a-8efab09bcf8a" />
<img width="1919" height="951" alt="Screenshot 2026-04-03 113118" src="https://github.com/user-attachments/assets/92d59cf5-6afc-4d52-86f2-a798ccedb8ae" />
<img width="1919" height="963" alt="Screenshot 2026-04-03 113453" src="https://github.com/user-attachments/assets/c87741af-61a2-404e-bcd7-5d963d6f3cc5" />

Anusha-Gali reviewed (approved):
Hi Namitha,

Thank you so much for proactively fixing the issues. Everything appears to be as per requirement.

<img width="1274" height="760" alt="Image" src="https://github.com/user-attachments/assets/0a774736-a4d1-4b23-8a84-efa9905af045" />
<img width="1274" height="760" alt="Image" src="https://github.com/user-attachments/assets/13bdbadb-0dce-4cff-bd87-14cb0b624586" />
<img width="1274" height="760" alt="Image" src="https://github.com/user-attachments/assets/aab2391e-d426-4161-9970-36f8a1b6c699" />

one-community commented:
Thank you all, merging!
